ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/idam/services/UserService.java

```diff
@@ -13,40 +18,81 @@
     private final IdamCaseworkerProperties idamCaseworkerProperties;
+    private final Oauth2 oauth2;
 
     @Autowired
-    public UserService(IdamApi idamApi, IdamCaseworkerProperties idamCaseworkerProperties) {
+    public UserService(
+        IdamApi idamApi,
+        IdamCaseworkerProperties idamCaseworkerProperties,
+        Oauth2 oauth2
+    ) {
         this.idamApi = idamApi;
         this.idamCaseworkerProperties = idamCaseworkerProperties;
+        this.oauth2 = oauth2;
+    }
+
+    public UserDetails getUserDetails(String authorisation) {
+        return idamApi.retrieveUserDetails(authorisation);
+    }
+
+    public User authenticateSystemUpdateUser(){
+        IdamCaseworker system = idamCaseworkerProperties.getSystem();
+        return authenticateUser(system.getUsername(), system.getPassword());
     }
 
     public User getUser(String authorisation) {
         return new User(authorisation, idamApi.retrieveUserDetails(authorisation));
     }
 
-    public String authenticateUser(String username, String password) {
-        AuthenticateUserResponse authenticateUserResponse = idamApi
-            .authenticateUser(getBasicAuthHeader(username, password));
+    public User authenticateUser(String username, String password) {
 
-        return BEARER + authenticateUserResponse.getAccessToken();
+        String authorisation = getIdamOauth2Token(username, password);
+        UserDetails userDetails = idamApi.retrieveUserDetails(authorisation);
+        return new User(authorisation, userDetails);
     }
 
-    public String authenticateSystemUpdateUser() {
-        IdamCaseworker systemUpdateUser = idamCaseworkerProperties.getSystemUpdateUser();
-        return authenticateUser(systemUpdateUser.getUsername(), systemUpdateUser.getPassword());
+    public User authenticateAnonymousCaseWorker() {
+        IdamCaseworker anonymousCaseworker = idamCaseworkerProperties.getAnonymous();
+        return authenticateUser(anonymousCaseworker.getUsername(), anonymousCaseworker.getPassword());
     }
 
-    public String authenticateAnonymousCaseworker() {
-        IdamCaseworker anonymousCaseworker = idamCaseworkerProperties.getAnonymousCaseworker();
-        return authenticateUser(anonymousCaseworker.getUsername(), anonymousCaseworker.getPassword());
+    public GeneratePinResponse generatePin(String name, String authorisation) {
+        return idamApi.generatePin(new GeneratePinRequest(name), authorisation);
     }
 
-    private String getBasicAuthHeader(String username, String password) {
-        String auth = username + ":" + password;
-        return "Basic " + new String(Base64.getEncoder().encode(auth.getBytes()));
+    public String getBasicAuthHeader(String username, String password) {
+        String authorisation = username + ":" + password;
+        return BASIC + Base64.getEncoder().encodeToString(authorisation.getBytes());
     }
 
+    public String getIdamOauth2Token(String username, String password) {
```

Comment:
if you're making changes here, you should look at using this instead:
https://github.com/hmcts/idam-java-client

Reply:
https://github.com/hmcts/idam-java-client/blob/master/src/main/java/uk/gov/hmcts/reform/idam/client/IdamClient.java#L32